### ISSUE TITLE
Opening Transpose Dialog does not set focus properly

### ISSUE THREAD

**Issue by jmwmusic** [musescore/MuseScore#31812]
### Issue type

Accessibility issue (e.g. for keyboard-only or screen reader users)

### Description with steps to reproduce

Steps:
1. Select some score to transpose.
2. From the Tools menu, select Transpose.
The Transpose dialog appears, but screen reader just announces the name of the dialog and not a control in the dialog.

Expected behaviour: on opening any dialog, focus should be set to the first control in the dialog.
In this case, Transpose Chromatically.

Workaround: Press Tab to regain focus.

### Supporting files, videos and screenshots

N/A

### In which versions of MuseScore Studio is this issue present?

4.6

### Regression

I was unable to check

### Operating system

Windows 11 with NVDA

### Checklist

- [x] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [x] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [x] I have attached all requested files and information to this report
- [x] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible
